RedisRepo.IntegrationTests/RedisTests.cs

```diff
@@ -415,6 +417,97 @@ public void Append_Init()
 
         #endregion
 
+        #region Redis Sets
+
+        [TestMethod]
+        public void TestAddToRedisSet()
+        {
+            const string key = "k1";
+            var values = new[] { "bar", "bar", "a", "b", "c" };
+
+            redisContext.AddToSet(key, values);
+
+            ValidateSetResults(key, new []{"a", "b","c", "bar"});   
+        }
+
+        [TestMethod]
+        public void TestCountTestMembers()
+        {
+            const string key = "k2";
+            var values = new[] { "bar", "bar", "a", "b", "c", "a", "b" };
+
+            redisContext.AddToSet(key, values);
+            
+            Assert.AreEqual(4, redisContext.CountSetMembers(key));
+        }
+
+        [TestMethod]
+        public void TestSetsDiff()
+        {
+            const string key1 = "testSetDiff1";
+            var values1 = new[] { "a", "b", "c", "d", "e" };
+
+            const string key2 = "testSetDiff2";
+            var values2 = new[] { "a", "b", };
+
+            redisContext.AddToSet(key1, values1);
+            redisContext.AddToSet(key2, values2);
+
+            var results = redisContext.GetSetsDifference(new[] {key1, key2});
+            AssertArraysRepresentSameSet(new [] {"c","d", "e"}, results);
+        }
+
+        [TestMethod]
+        public void TestSetsUnion()
+        {
+            const string key1 = "TestSetsUnion1";
+            var values1 = new[] { "a", "c" };
+
+            const string key2 = "TestSetsUnion2";
+            var values2 = new[] { "a", "b" };
+
+            redisContext.AddToSet(key1, values1);
+            redisContext.AddToSet(key2, values2);
+
+            var results = redisContext.UnionSets(new[] { key1, key2 });
+            AssertArraysRepresentSameSet(new[] { "a", "b", "c" }, results);
+        }
+
+        [TestMethod]
+        public void TestSetsIntersect()
+        {
+            const string key1 = "TestSetsIntersect1";
+            var values1 = new[] { "a", "c" };
+
+            const string key2 = "TestSetsIntersect2";
+            var values2 = new[] { "a", "b" };
+
+            redisContext.AddToSet(key1, values1);
+            redisContext.AddToSet(key2, values2);
+
+            var results = redisContext.IntersectSets(new[] { key1, key2 });
+            AssertArraysRepresentSameSet(new[] { "a" }, results);
+        }
+
+        private void ValidateSetResults(string key, string[] expected)
+        {
+            var valuesFromRedis = redisContext.GetSetMembers(key);
+            AssertArraysRepresentSameSet(expected, valuesFromRedis);
+        }
+
+        private static void AssertArraysRepresentSameSet(string[] expected, string[] valuesFromRedis)
```

Comment:
Use existing Collections.Assert(), also, this will pass for { "a", "a", "b" } vs. { "a", "b", "c" } which is not the intent